awx/main/access.py

```diff
@@ -645,22 +645,24 @@ def can_delete(self, obj):
         return False
 
     def can_attach(self, obj, sub_obj, relationship, *args, **kwargs):
-        if not settings.MANAGE_ORGANIZATION_AUTH and not self.user.is_superuser:
-            return False
-
         # Reverse obj and sub_obj, defer to RoleAccess if this is a role assignment.
         if relationship == 'roles':
             role_access = RoleAccess(self.user)
             return role_access.can_attach(sub_obj, obj, 'members', *args, **kwargs)
-        return super(UserAccess, self).can_attach(obj, sub_obj, relationship, *args, **kwargs)
 
-    def can_unattach(self, obj, sub_obj, relationship, *args, **kwargs):
         if not settings.MANAGE_ORGANIZATION_AUTH and not self.user.is_superuser:
             return False
```

Comment:
I'm staring at this hard, and right now I think that this remaining `if` clause is superfluous.

I ask myself the question "what can you attach to a user?". To answer this I look at the user API "related" items. I see organizations and teams, but these views do not allow attaching. Some other views allow creation, but not attachment.

In fact, I don't think this line should ever be hit. If you're concerned about being conservative, I'd prefer to raise a RuntimeError instead of the call to super.

Reply:
That does seem likely, doesn't it?  Since group membership seems to be encoded using roles like `Organization.member_role`, etc.  I think I like raising `RuntimeError`, just to be sure.

Reply:
I ultimately went with logging and returning False, as we were already doing in `RoleAccess.can_unattach` in a similar case.